### PR TITLE
Merge main into v2

### PR DESCRIPTION
**Pull request by Trivo25**
all the goods 

most of the PR merges main into v2; some additional v2 prep is happening here https://github.com/o1-labs/o1js/pull/1873/commits/33b7c6adff44e7b38dcb9bd17b2d6a8a15aafcd6

### CODE REVIEW

File: tests/vk-regression/vk-regression.json [hunk 1]
```diff
@@ -100,21 +100,21 @@
...
     }
   },
   "Dex": {
-    "digest": "c4f3529db60868ed3a04dc48a8e09fbb9e5814755f479570dede6a3fb84821",
+    "digest": "fee88991bcc3331978a787642273ee02cbfc738949c6daad38c3bab65a8e81e",
     "methods": {
       "approveBase": {
-        "rows": 13292,
-        "digest": "851c0a9eb2d015d88703ef0d5cebf663"
+        "rows": 6175,
```
Comment (Trivo25) [inline tests/vk-regression/vk-regression.json:116]: expected because `TokenContractV2` supported way less account updates

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,19 +15,63 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+- `zkProgram` methods now support `auxiliaryOutput`. https://github.com/o1-labs/o1js/pull/1809
+  - Each program method now accepts an optional property `auxiliaryOutput`
+  - Auxiliary output is additional output that the zkProgram method returns
+- New method `toCanonical()` in the `Provable<T>` interface to protect against incompleteness of certain operations on malicious witness inputs https://github.com/o1-labs/o1js/pull/1759
+- `divMod64()` division modulo 2^64 that returns the remainder and quotient of the operation
+- `addMod64()` addition modulo 2^64
+- Bitwise OR via `{UInt32, UInt64}.or()`
+- **BLAKE2B hash function** gadget [#1285](https://github.com/o1-labs/o1js/pull/1285)
+
+## [1.9.0](https://github.com/o1-labs/o1js/compare/450943...f15293a69) - 2024-10-15
```
Comment (45930) [inline CHANGELOG.md:47]: Looks like the 1.9.1 entry was merged out

File: CHANGELOG.md [hunk 3]
```diff
@@ -15,19 +15,63 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
     _Security_ in case of vulnerabilities.
  -->
 
-## [Unreleased](https://github.com/o1-labs/o1js/compare/450943...HEAD)
+## [Unreleased](https://github.com/o1-labs/o1js/compare/f15293a69...HEAD)
+
+## [2.0.0](https://github.com/o1-labs/o1js/compare/f15293a69...HEAD)
+
+### Breaking Changes
```
Comment (45930) [inline CHANGELOG.md:22]: Do we have an upgrade guide to the breaking changes?

Comment (ymekuria) [reply CHANGELOG.md:22]: I don't believe we do, but it is a good idea.

Comment (Trivo25) [reply CHANGELOG.md:22]: see slack

File: CHANGELOG.md [hunk 4]
```diff
@@ -15,19 +15,70 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
...
+  - The return type of a proven method has changed as a result of this.
+- Various breaking constraint changes in internal methods or circuits because of audit fix.
+- Removal of various deprecated methods and functions.
+  - Promotion of various methods and functions to stable as part of change.
+  - A slightly modified encryption and decryption algorithm. https://github.com/o1-labs/o1js/pull/1729
+- Promotion of `TokenContractV2` to `TokenContract` with a correct amount of maximum account updates.
+
+### Added
+
+- `zkProgram` methods now support `auxiliaryOutput`. https://github.com/o1-labs/o1js/pull/1809
```
Comment (ymekuria) [inline CHANGELOG.md:38]: nit: Should this `ZkProgram`? I'm not actually sure what the correct way is so this is more a question to align.

**Review by ymekuria — APPROVED** [review]
LGTM.

File: CHANGELOG.md [hunk 5]
```diff
@@ -15,7 +15,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
     _Security_ in case of vulnerabilities.
  -->
 
-## [Unreleased](https://github.com/o1-labs/o1js/compare/f15293a69...HEAD)
+## [Unreleased](https://github.com/o1-labs/o1js/compare/7e9394...HEAD)
 
 ## [2.0.0](https://github.com/o1-labs/o1js/compare/f15293a69...HEAD)
```
Comment (45930) [inline CHANGELOG.md:20]: [NIT]

I think this compare should use `7e9394` as the base, rather than `f15293a69` which comes from v1.9.0

**Review by 45930 — APPROVED** [review]

File: CHANGELOG.md (hunk 5 above)
Comment (Trivo25) [reply CHANGELOG.md:20]: Yep! Still need to shuffle commits around, I'll pick the final commit in a separate release PR

**Pull request merged**